src/binaryen-c.h

```diff
@@ -103,6 +136,17 @@ typedef void* BinaryenFunctionTypeRef;
 // Add a new function type. This is thread-safe.
 // Note: name can be NULL, in which case we auto-generate a name
 BinaryenFunctionTypeRef BinaryenAddFunctionType(BinaryenModuleRef module, const char* name, BinaryenType result, BinaryenType* paramTypes, BinaryenIndex numParams);
+// Gets the function type at the specified index. Returns `NULL` when there are no more function types.
```

Comment:
why do you want to get function types by their index? can't you use their names?

Reply:
Idea here is to have a simple way to iterate over all function types (by calling this function with 0, 1, etc. until it returns `NULL`) and then being able to use the getters and setters below to evaluate and maybe modify them. Could be used, for example, when loading an existing module with the intention to unify function type names to combinations of "i", "I", "f", "F" and "v".

Reply:
I see. For iteration, perhaps another option is something like a JS API that returns an array of the names, and a C API underneath that does something similar (mallocs a list that must later be freed, and returns it)? Just a suggestion, I'm not sure what is best, but overall I'd like to avoid using indexes (as we have names intentionally, and also, the indexes are internal and do not necessarily match the wasm binary indexes, which could be confusing).

If you want, you can split this into smaller PRs, as we can land the obvious stuff already, but the areas where the API design isn't yet clear may take longer.

Reply:
Sure, going to split it up then - and taking the array idea into account.